src/kernel/filesystem/vfs.zig

```diff
@@ -560,8 +561,8 @@ test "mount" {
 
     // The root fs
     var testfs = try testInitFs(allocator);
-    defer testfs.deinit();
     defer allocator.destroy(testfs);
+    defer testfs.deinit();
```

Comment:
I think it still needs to be `deinit` then `free` as when freeing first, will be deiniting a free pointer

Reply:
I was getting seg faults when using the previous order and upon looking at the docs I read that the deferred statements are run in reverse order. which is why I defer deinitting the testfs last.

Reply:
Ah, ok, I thought it was the other way, so I will have to re-order some of mine then

Reply:
Yeah me too! It actually makes sense to have it reversed though.